src/codegen/mod.rs

```diff
@@ -561,7 +561,13 @@ impl CodeGenerator for Type {
                     let layout = self.layout(ctx).unwrap_or_else(Layout::zero);
                     BlobTyBuilder::new(layout).build()
                 } else {
-                    inner_item.to_rust_ty(ctx)
+                    let inner_rust_ty = inner_item.to_rust_ty(ctx);
+                    if inner_rust_ty == aster::AstBuilder::new().ty().unit() {
```

Comment:
This conditional is hacky enough it deserves at least a comment. But I really think we should just check the type to see if it's a template instantiation? Or at least, given `to_rust_ty` should never generate `unit` types unless that, what about `debug_assert!`ing instead?